elephant/spectral.py

```diff
@@ -627,11 +417,15 @@ def welch_cohere(x, y, num_seg=8, len_seg=None, freq_res=None, overlap=0.5,
     params['nperseg'] = nperseg
     params['noverlap'] = int(nperseg * overlap)
 
-    freqs, Pxy = _welch(xdata, ydata, **params)
-    freqs, Pxx = _welch(xdata, xdata, **params)
-    freqs, Pyy = _welch(ydata, ydata, **params)
-    coherency = np.abs(Pxy)**2 / (np.abs(Pxx) * np.abs(Pyy))
-    phase_lag = np.angle(Pxy)
+    freqs, Pxx = scipy.signal.welch(xdata, **params)
+    _, Pyy = scipy.signal.welch(ydata, **params)
+    _, Pxy = scipy.signal.csd(xdata, ydata, **params)
+
+    coherency = np.abs(Pxy) ** 2 / (Pxx * Pyy)
+
+    # conjugate is taken to follow the old behavior of _welch() function
+    # that was a copy-paste of scipy v0.16.0
+    phase_lag = np.angle(Pxy.conj())
```

Comment:
It seems that our original implementation of _welch() was wrong.
Please remove this conjugate.

Reply:
then we should change the ground truth of one of the tests
remove the conjugate and try running tests

Reply:
I didn't find a test where the actual values of phase difference were tested (existing tests are always comparison between the results from different usages of the same function).
Where is it?

Reply:
fixed the ground truth